src/lasagna/easy.py

```diff
@@ -0,0 +1,162 @@
+from typing import (
+    Callable,
+    List,
+    Type,
+    Any, # FIXME
+)
+import json
+
+from pydantic import BaseModel
+
+from . import (
+    build_simple_agent,
+    Model,
+    EventCallback,
+    EventPayload,
+    AgentRun,
+    ExtractionType,
+    flat_messages,
+    noop_callback,
+    recursive_extract_messages,
+)
+
+
+def simple_text_callback_binder(simple_text_callback: Callable[[str], None]) -> Callable[[EventPayload], None]:
+    async def bound_simple_text_callback(event: EventPayload):
+        if event[0] == 'ai' and event[1] == 'text_event':
+            await simple_text_callback(event[2])
+    return bound_simple_text_callback
+
+
+async def simple_ask(
+    partially_bound_model: Any,
```

Comment:
I'd change all these `partially_bound_model: Any` to:
> `binder: Callable[[AgentCallable], BoundAgentCallable]`

Which btw is totally _not obvious_ since I don't have docs, so that's why I'm saying it here.